include/RAJA/pattern/detail/reduce.hpp

```diff
@@ -93,26 +103,38 @@ struct max : detail::op_adapter<T, RAJA::operators::maximum> {
 namespace detail
 {
 
-template <typename T, bool doing_min = true>
+template <typename T>
+struct DefaultLoc
+{
+  RAJA_HOST_DEVICE constexpr T value() { return T(); }
+};
+
+template <>
+struct DefaultLoc<Index_type>
```

Comment:
probably want to use std::is_integral to define a class of specializations

Reply:
Addressed in latest commits.